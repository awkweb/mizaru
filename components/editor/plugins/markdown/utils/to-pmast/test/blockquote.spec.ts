/* eslint-disable import/default */
import u from 'unist-builder'

import toPMAST from '../'

test('empty', () => {
    const mdast = u('root', [
        u('blockquote', [u('paragraph', [u('syntax', '> ')])]),
    ])
    const pmast = toPMAST(mdast)
    expect(pmast).toMatchInlineSnapshot(`
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "> ",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "blockquote",
            },
          ],
          "type": "doc",
        }
    `)
})

for (const syntax of ['>', '> ']) {
    describe(`"${syntax}"`, () => {
        test('single word', () => {
            const mdast = u('root', [
                u('blockquote', [
                    u('paragraph', [u('syntax', syntax), u('text', 'foo')]),
                ]),
            ])
            const pmast = toPMAST(mdast)
            expect(pmast).toMatchSnapshot()
        })

        test('multiple words', () => {
            const mdast = u('root', [
                u('blockquote', [
                    u('paragraph', [
                        u('syntax', syntax),
                        u('text', 'foo bar baz'),
                    ]),
                ]),
            ])
            const pmast = toPMAST(mdast)
            expect(pmast).toMatchSnapshot()
        })

        test('split across multiple lines', () => {
            const mdast = u('root', [
                u('blockquote', [
                    u('paragraph', [u('syntax', syntax), u('text', 'foo')]),
                    u('paragraph', [u('text', 'bar')]),
                    u('paragraph', [u('text', 'baz')]),
                ]),
            ])
            const pmast = toPMAST(mdast)
            expect(pmast).toMatchSnapshot()
        })
    })
}
