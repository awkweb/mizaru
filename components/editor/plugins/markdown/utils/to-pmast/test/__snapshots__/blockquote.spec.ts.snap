// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"> " multiple words 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "> ",
              "type": "text",
            },
            Object {
              "text": "foo bar baz",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`"> " single word 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "> ",
              "type": "text",
            },
            Object {
              "text": "foo",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`"> " split across multiple lines 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "> ",
              "type": "text",
            },
            Object {
              "text": "foo",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "content": Array [
            Object {
              "text": "bar",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "content": Array [
            Object {
              "text": "baz",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`">" multiple words 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": ">",
              "type": "text",
            },
            Object {
              "text": "foo bar baz",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`">" single word 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": ">",
              "type": "text",
            },
            Object {
              "text": "foo",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`">" split across multiple lines 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": ">",
              "type": "text",
            },
            Object {
              "text": "foo",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "content": Array [
            Object {
              "text": "bar",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "content": Array [
            Object {
              "text": "baz",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;
